### PR TITLE
swap events

### PR DESCRIPTION
**Pull request by roeierez**

### CODE REVIEW

File: libs/sdk-core/src/models.rs
Comment (dangeross) [inline libs/sdk-core/src/models.rs:1228]: Is there any need to migrate the swap_info table to update the swap status from Expired to Refundable/Completed else it will become WaitingConfirmation

Comment (roeierez) [reply libs/sdk-core/src/models.rs:1228]: Sure, I am waiting with the migration until I'll finish the implementation.
My plan is to reset all of them to Initial which will include them in the next rescan by the swapper. It will update the right state for all of them.

File: libs/sdk-core/src/breez_services.rs
Comment (ok300) [inline libs/sdk-core/src/breez_services.rs:109]:
```suggestion
    /// Indicates that this swap's status was just updated
```

This more strongly indicates that mainly the status has changed, and not only just some of the fields.

File: libs/sdk-core/src/chain.rs
Comment (ok300) [inline libs/sdk-core/src/chain.rs:85]:
```suggestion
```

IMO better to remove this comment, as the method doesn't only deal with confirmed utxos anymore.

File: libs/sdk-core/src/models.rs
Comment (ok300) [inline libs/sdk-core/src/models.rs:1362]: Incomplete comment?

File: libs/sdk-core/src/persist/swap.rs
Comment (ok300) [inline libs/sdk-core/src/persist/swap.rs:108]: Unwrap is bad, as it can panic on failure.

File: libs/sdk-core/src/swap_in/swap.rs
Comment (ok300) [inline libs/sdk-core/src/swap_in/swap.rs:92]: Since we already have the persister, it might be simpler to use
```rust
self.persister.node_info()?.block_height
```

like in https://github.com/breez/breez-sdk/pull/812/files#diff-dfcf1c8c08fb243da9a4e02f631600d2e5a5c827fbbed56c8e98daadd4347483R253

Comment (roeierez) [reply libs/sdk-core/src/swap_in/swap.rs:92]: Good point but when tested I have noticed the greenlight block sometimes lag behind the mempool space block.
Almost always there is a 1 block lag. Also when the node is not connected for a while it takes some time to catchup and gl shows the tip it is aware of which lags a lot behind for a while.
Perhaps if gl will expose a listener to block change it will make more sense as even if lags a bit behind it will be consistent. What do you think?

File: libs/sdk-core/src/chain.rs
Comment (roeierez) [reply libs/sdk-core/src/chain.rs:79]: Actually in case where this flag is set to true,  the result will not include confirmed utxos if they were spend by unconfirmed transactions.
This is because we consider these spends even though they are not confirmed

File: libs/sdk-core/src/breez_services.rs
Comment (roeierez) [reply libs/sdk-core/src/breez_services.rs:109]: I combined your suggestion in the comment as we we emit events not only when the status has changed.

File: libs/sdk-core/src/chain.rs
Comment (roeierez) [reply libs/sdk-core/src/chain.rs]: removed

File: libs/sdk-core/src/persist/swap.rs
Comment (roeierez) [reply libs/sdk-core/src/persist/swap.rs]: True, made it in by mistake, we don't need it.

File: libs/sdk-core/src/models.rs
Comment (roeierez) [reply libs/sdk-core/src/models.rs]: Fixed

File: libs/sdk-core/src/chain.rs
Comment (dangeross) [reply libs/sdk-core/src/chain.rs:79]: Apologies, following the code further down it makes sense

File: libs/sdk-core/src/swap_in/swap.rs
Comment (ok300) [reply libs/sdk-core/src/swap_in/swap.rs:92]: I wasn't aware the GL tip was lagging. Yes, then it makes sense the way you did it.

I'll abandon my other PR using `self.persister.node_info()?.block_height` as it would suffer from the same problem.

File: libs/sdk-core/src/breez_services.rs
Comment (ok300) [reply libs/sdk-core/src/breez_services.rs:109]: I see, then I misunderstood it. I thought the event is only emitted on status change.

Comment (ok300) [reply libs/sdk-core/src/breez_services.rs:109]: May I ask what's the use case for that? Would a user want to get an event for something other than a status change?

Comment (roeierez) [reply libs/sdk-core/src/breez_services.rs:109]: Sure, among other things we update the swap chain information such as refund tx id (after refund).
We also update the swap open channel fees from time to time.

Comment (ok300) [reply libs/sdk-core/src/breez_services.rs:109]: Makes sense, thanks.

**Review by ok300 — APPROVED** [review]

**Review by dangeross — APPROVED** [review]
LGTM, pending migrations

**Comment by erdemyerebasmaz** [comment]
@roeierez
Documentation for SwapInfo.status should be updated to reflect these changes. [L1275-R1291](https://github.com/breez/breez-sdk/pull/811/files#diff-78b2d03eb877d309a4fe8572e1599cc8918d7879dfdbb88ddae440a556fce9fbL1275-R1291)

**Pull request merged**
